packages/base/src/sap/ui/webcomponents/base/Configuration.js

```diff
@@ -1,30 +1,243 @@
+
+const detectLanguage = () => {
+	const browserLanguages = navigator.languages;
+
+	const navigatorLanguage = () => {
+		if (Device.os.android) {
+			// on Android, navigator.language is hardcoded to 'en', so check UserAgent string instead
+			const match = navigator.userAgent.match(/\s([a-z]{2}-[a-z]{2})[;)]/i);
+			if (match) {
+				return match[1];
 			}
+			// okay, we couldn't find a language setting. It might be better to fallback to 'en' instead of having no language
+		}
+		return navigator.language;
+	};
+
+	const rawLocale = (browserLanguages && browserLanguages[0]) || navigatorLanguage() || navigator.userLanguage || navigator.browserLanguage;
+
+	return rawLocale || "en";
+};
+
+const language = detectLanguage();
+
+const CONFIGURATION = {
+	theme: "sap_fiori_3",
+	rtl: null,
+	language: new Locale(language),
+	compactSize: false,
+	supportedLanguages,
+	calendarType: null,
+	derivedRTL: null,
+	"xx-wc-force-default-gestures": false,
+	"xx-wc-no-conflict": false, // no URL
+};
+
+const formatSettings = new FormatSettings(CONFIGURATION);
+
+/* General settings */
+const getTheme = () => {
+	return CONFIGURATION.theme;
+};
+
+const getRTL = () => {
+	return CONFIGURATION.rtl === null ? CONFIGURATION.derivedRTL : CONFIGURATION.rtl;
+};
+
+const getLanguage = () => {
+	return CONFIGURATION.language.sLocaleId;
+};
+
+const getCompactSize = () => {
+	return CONFIGURATION.compactSize;
+};
+
+const getSupportedLanguages = () => {
+	return CONFIGURATION.supportedLanguages;
+};
+
+/* WC specifics */
+const getWCForceDefaultGestures = () => {
+	return CONFIGURATION["xx-wc-force-default-gestures"];
+};
+
+const getWCNoConflict = () => {
+	return CONFIGURATION["xx-wc-no-conflict"];
+};
+
+/* Calendar stuff */
+const getCalendarType = () => {
+	if (CONFIGURATION.calendarType) {
+		const type = Object.keys(CalendarType).filter(calType => calType === CONFIGURATION.calendarType)[0];
+
+		if (type) {
+			return type;
+		}
+	}
+
+	return LocaleData.getInstance(getLocale()).getPreferredCalendarType();
+};
+
+const getOriginInfo = () => {};
+
+const getLocale = () => {
+	return CONFIGURATION.language;
+};
+
+const getFormatSettings = () => {
+	return formatSettings;
+};
+
+const _setTheme = themeName => {
+	CONFIGURATION.theme = themeName;
+};
+
+const applyURLParameters = () => {
+	const fullURL = window.location.href;
+	const params = new URL(fullURL);
+	const searchParams = params.searchParams;
+	const langValue = searchParams.get("sap-ui-language");
+	const theme = searchParams.get("sap-ui-theme");
+	const rtl = searchParams.get("sap-ui-rtl");
+	const compactSize = searchParams.get("sap-ui-compactSize");
+	const calendarType = searchParams.get("sap-ui-calendarType");
+	const forceDefaultGestures = searchParams.get("xx-wc-force-default-gestures");
+
+	if (langValue) {
+		setLanguage(langValue);
+	}
+
+	if (theme) {
+		CONFIGURATION.theme = theme;
+	}
+
+	if (rtl !== null) {
```

Comment:
can we extract this in a method for all booleans and also make it case insensitive